homeassistant/components/freebox/sensor.py

```diff
@@ -112,16 +113,47 @@ def should_poll(self) -> bool:
         """No polling needed."""
         return False
 
+    @callback
     async def async_on_demand_update(self):
         """Update state."""
         self.async_schedule_update_ha_state(True)
```

Comment:
Drop `async`, as it's now a callback.

Rename `async_update` to `async_update_state` and make that a callback.

```suggestion
    async def async_on_demand_update(self):
        """Update state."""
        self.async_update_state()
        self.async_write_ha_state()
```

And also add `self.async_update_state()` to `async_added_to_hass`